### ISSUE TITLE
Arrow function name doesn't have same highlight as regular function name

### ISSUE THREAD

**Issue by maharjanaman** [pangloss/vim-javascript#1247]
Shouldn't highlight for arrow function name be same as normal function name?

![Screen Shot 2022-06-03 at 09 28 47](https://user-images.githubusercontent.com/14164484/171781874-f802a289-6e2e-4d49-ad9f-d81deffc2abb.png)

Here `sum` and `sumOne` has different highlights.

**Comment by amadeus** [comment]
They are technically two different things. The first one is a function declaration -- the name `sum` in this case has special properties -- you can put that function declaration anywhere in the file (even as the last written code) but still use it anywhere (even on the first line of the file)

The second example is a function expression -- `sumOne` is not usable until AFTER `sumOne` is declared. This is also similar to how any other `const` or `let` works.

Anything after a `const` or a `let` -- is impossible for us to know what it is.

Take this crazy contrived example:

```js
const sumOne = (() => {
  if (someValue) {
    return () => {};
  }
  return 2;
})()
```

`sumOne` could be either the number 2 or a function, which could be determined at runtime.

In other words, it's pretty out of scope for use to do any sort of highlighting that would require a language server to parse the file and figure out types for us.
